### PR TITLE
Collect jira issue type for new Vulnerability issue type.

### PR DESCRIPTION
**Pull request by jsvob**
Related to OSIDB-2980
Closes OSIDB-3240

### CODE REVIEW

**Comment by jsvob** [comment]
This should ideally wait for merge of https://github.com/RedHatProductSecurity/osidb/pull/678 and then I'll rebase/rework it based on that.

File: collectors/jiraffe/tests/test_collectors.py [hunk 1]
```diff
@@ -374,17 +374,32 @@ def test_collect_tracker_with_multi_cve_flaw(self):
...
+    @pytest.mark.parametrize(
+        "project_key,fields_count,vulntype_exists",
+        [
+            ("RHEL", 125, False),
+            ("OSIM", 18, False),
+            ("RHEL", 135, True),
+            ("OSIM", 19, True),
+        ],
+    )
+    def test_collect(self, pin_envs, project_key, fields_count, vulntype_exists):
```
Comment (JakubFrejlach) [inline collectors/jiraffe/tests/test_collectors.py:395]: `vulntype_exists` is a preparation for future work? I don't see it used anywhere.

**Review by JakubFrejlach — APPROVED** [review]
Couple of small things but otherwise LGTM!

**Review by osoukup — APPROVED** [review]
LGTM

File: collectors/jiraffe/tests/test_collectors.py (hunk 1 above)
Comment (jsvob) [reply collectors/jiraffe/tests/test_collectors.py:395]: The value is not used anywhere, but is used to have a second set of test cassettes, to test the different behavior:

```
    # When re-recording VCRs, for vulntype_exists=False, copy the cassettes recorded for
    # vulntype_exists=True and manually tweak the cassette file to simulate the Vulnerability
    # type not existing. This visibly highlights the behavior of MetadataCollector and makes
    # it visible if there's a bug where all fields of type 1 are discarded when type 12207
    # is collected (an actual bug experienced during development).
    # If it works correctly, fields_count with vulntype_exists=True must not be lower than with
    # vulntype_exists=False.
```

Do you have a suggestion how to make it more clear?

File: collectors/jiraffe/collectors.py [hunk 2]
```diff
@@ -239,7 +239,40 @@ def free_queriers(self):
...
+
+        try:
+            res = self.jira_querier.jira_conn._get_json("issuetype")
+            for t in res:
+                if t["name"] == "Vulnerability":
+                    id_of_vulnerability_issue_type = t["id"]
+                    break
+        except JIRAError as e:
+            if e.status_code == 400:
+                print(e.response)
```
[the remark this answers is not in the document]
Comment (JakubFrejlach) [reply collectors/jiraffe/collectors.py:264]: Nevermind, I somehow assumed this is your code, I've actually touched this collector in my last PR and removed the print myself. Print is useless as it only prints the response object.

[the remark this answers is not in the document]
Comment (jsvob) [reply collectors/jiraffe/collectors.py:264]: Aha, thanks, I'll remove it too when rebasing, then :)

File: collectors/jiraffe/tests/test_collectors.py (hunk 1 above)
Comment (jsvob) [reply collectors/jiraffe/tests/test_collectors.py:409]: Updated the comment to be clearer.

**Comment by jsvob** [comment]
@JakubFrejlach Any idea why tests worked in https://github.com/RedHatProductSecurity/osidb/pull/678 but the same tests don't work now? I didn't touch the tests edited in https://github.com/RedHatProductSecurity/osidb/pull/678, yet those are the tests that fail now.

File: collectors/jiraffe/tests/test_collectors.py (hunk 1 above)
Comment (JakubFrejlach) [reply collectors/jiraffe/tests/test_collectors.py:409]: I see. Probably I don't have any better solution, keep this, I think it is clear enough.

**Comment by JakubFrejlach** [comment]
> @JakubFrejlach Any idea why tests worked in #678 but the same tests don't work now? I didn't touch the tests edited in #678, yet those are the tests that fail now.

@jsvob I was using the same cassetess for my two tests I've added in #678, you can reuse the cassetess:
https://github.com/RedHatProductSecurity/osidb/blob/master/collectors/jiraffe/tests/test_collectors.py#L443
https://github.com/RedHatProductSecurity/osidb/blob/master/collectors/jiraffe/tests/test_collectors.py#L411
https://github.com/RedHatProductSecurity/osidb/blob/master/collectors/jiraffe/tests/test_collectors.py#L388

I see you've also removed the cassete specification in the first test, I wanted to make it explicit when I've already made the cassetes shared between 3 tests

**Comment by JakubFrejlach** [comment]
> @JakubFrejlach Thank you. Turns out, I also made a second mistake: The collector now makes an extra request and that isn't in the original cassettes. To keep as much continuity as possible to show that the core function of the collector has not been compromised, I manually added the request&response for `issuetype` so that the original request in the original cassettes is left untouched. The new cassettes contain actual recorded traffic.

Thank you! LGTM now!

**Pull request merged**